datacache/cache.py

```diff
@@ -16,10 +16,11 @@ def __init__(self, subdir="datacache"):
         self.local_paths = {}
 
     def delete_key(self, url):
-        path = self.local_paths(url)
-        del self.local_paths[url]
-        if exists(path):
-            remove(path)
+        if url in self.local_paths:
+            path = self.local_paths[url]
+            del self.local_paths[url]
+            if exists(path):
+                remove(path)
```

Comment:
It's probably worth distinguishing `local_path` the method, `local_paths` the dictionary, since that seems to be typo I made originally. Maybe `self._local_paths_dict`?